### DISCUSSION TITLE
Adding 'title' in the API throws an error

### DISCUSSION THREAD

**Discussion by tacman** [biati-digital/glightbox#519] [Q&A] [answered]
This is pretty much from the documentation, but throws an error.

```html
    <script type="module">
        import 'https://cdn.jsdelivr.net/gh/mcstudios/glightbox/dist/js/glightbox.min.js';

        var exampleApi = GLightbox({ selector: null });
        exampleApi.insertSlide( {
            'href': 'https://picsum.photos/1200/800',
            'type': 'image',
            'title': 'My Title',
            'description': 'Example',
        });
        exampleApi.open();
</script>
```

throws an error:

stimulus.index-b5b1d00e42695b8959b4a1e94e3bc92a.js:7 application #start
glightbox.min.js:1 Uncaught TypeError: v.getAttribute is not a function
    at e.value (glightbox.min.js:1:52654)
    at glightbox.min.js:1:41042
    at glightbox.min.js:1:22791
    at HTMLImageElement.<anonymous> (glightbox.min.js:1:14623)

glightbox.min.js:1 Uncaught TypeError: v.getAttribute is not a function
    at e.value (glightbox.min.js:1:52654)
    at glightbox.min.js:1:41042
    at glightbox.min.js:1:22791
    at HTMLImageElement.<anonymous> (glightbox.min.js:1:14623)

**Comment by tacman** [comment]
Hmm, I thought discussions were for "how to", isn't this a bug?  Or an error in the documentation?  Or am I doing something wrong?

**Reply by gingerchew** [reply]
I converted it to a discussion because it was formatted like a discussion. There isn't a minimal reproduction, steps to reproduce, can you reproduce it on the demo site, all questions that we ask in the bug report issue template.

Is there a reason you're using the `insertSlide` function instead of a selector?

**Reply by tacman** [reply]
A selector is for when the data is embedded in the HTML, right? But my data comes from a json feed, so that's why I'm doing it all in javascript.

I believe you can cut and paste that snippet into any HTML page and see the error.

**Reply by gingerchew** [reply, marked as answer]
[Here is a codepen with just that.](https://codepen.io/gingerchew/pen/xxodLyX/dcd6470c5efeea8d348c6d581a9294e2) You can see that the snippet works fine and there is no error.

**Reply by tacman** [reply]
I must have a conflict with how I'm loading something, probably has to do with importmap or stimulus.  Thanks!

**Comment by tacman** [comment]
Oh, or is this because v4 is close to release?  Can you give an update on that?  If it's available on a CDN perhaps we can start testing, checking the docs, etc.

**Comment by tacman** [comment]
I'm going to investigate this more, perhaps there's a conflict on my website.

**Reply by sydtv** [reply]
were you able to fix your problem? i have the exact same issue right now, if i just remove the title and description from my json item all works like expected

**Comment by tacman** [comment]
yes, I was able to fix it, but not sure about the exact cause.  Here's my code, running 3.3 from the jsdelivr cdn.

```javascript
                data['member'].forEach(product => {
                    const slideData = { href: product.thumbnail };
                    if (product.title)
                        slideData.title = product.title;
                    if (product.description)
                        slideData.description = product.description;
                    slideData.descPosition = 'right';
                    exampleApi.insertSlide(slideData);
                })
```

Any update on version 4, @gingerchew?  It's such a cool library, it'd be wonderful to have an updated release.
